### PR TITLE
Factor FW finding logic into separate script

### PR DESCRIPTION
**Pull request by sl-mlx**
The logic for finding firmware archives turns out to be useful outside
of bfvcheck, so factor it into its own script. For now, it will be used
in two places: bfvcheck, and firmware installation testing.

### CODE REVIEW

File: bfvcheck [hunk 1]
```diff
@@ -155,7 +83,7 @@ if [ $MLXFW_SUCCESS -eq 0 ]; then
 
     # If FW (Running) line exists, user updated firmware, but did
     # not power cycle, so output message further down
-    if echo "\"$mlxfw_out\"" | grep -q 'FW *(Running)' >/dev/null; then
+    if [ ! -z "$(echo "\"$mlxfw_out\"" | grep 'FW *(Running)' )" ]; then
```
Comment (lsun100) [inline bfvcheck]: simplify "! -z" to "-n"?

Comment (sl-mlx) [reply bfvcheck]: I accidentally undid a shellcheck fix here, see the second commit

File: bffindfw [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+# Finds the FW file on the system and sets $mlxfw accordingly.
+# Returns 1 if not found, 0 otherwise.
+find_fw_with_name () {
+    # find the firmware package on the system
+    mlxfw="/opt/mellanox/mlnx-fw-updater/firmware/${mlxfw_name}"
+
+    if [ -f $mlxfw ]; then
+        return 0
+    else
```
Comment (lsun100) [inline bffindfw:78]: Potential optimization:
Looks like the 'else' and 'if' can be removed, and only line 80 is needed for line 83~86

File: bffindfw [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        fi
+    fi
+
+    err "warn: falling back to legacy FW filename"
+    mlxfw_name="$BASE_FW_NAME"
+    if find_fw_with_name; then
+        return 0
+    fi
+
+    return 1
```
Comment (lsun100) [inline bffindfw:105]: Maybe "if  else  return 1"  to look more consistent than the function above?

File: man/bffindfw.8 [hunk 4]
```diff
@@ -0,0 +1,12 @@
+.TH BFFINDFW 8 "July 2020"
+.SH NAME
+bffindfw \- Print the path to network adapter FW for current platform
+.SH SYNOPSIS
+.B bffindfw
```
Comment (lsun100) [inline man/bffindfw.8:5]: Just a small comment:

It's ok for now. In the future when we need to add more utility code for FW, we could consider using a more generic name (such as bfhcafw) with command line arguments to do different things.

Comment (sl-mlx) [reply man/bffindfw.8:5]: I'll just rename it now to reduce potential pain in the future

File: bffindfw (hunk 3 above)
Comment (sl-mlx) [reply bffindfw:105]: Same here, checks now follow the same syntax as above

File: bffindfw (hunk 2 above)
Comment (sl-mlx) [reply bffindfw:78]: Changed all the checks to look the same

**Pull request merged**